nntrainer/layers/lstm.cpp

```diff
@@ -80,6 +81,11 @@ int LSTMLayer::initialize(Manager &manager) {
                                       WeightRegularizer::NONE, 1.0f, true);
   }
 
+  mem_cell =
+    std::make_shared<Var_Grad>(output_dim[0], true, true, "LSTM:mem_cell");
```

Comment:
Please try and use the `mem_cell` as a weight so that is memory is managed by the manager than managing it manually in the layer.

Reply:
this is not the memory which managed by the manager. It is layer variables. the manager managed variable is h, weight_hh, weight_xh and bias which includes all variables, f, g, i and o.

Reply:
I guess there should be an interface with the manager to request for memory, as required in this case. Added #1086 for this.